tests/test_signal.py

```diff
@@ -46,3 +48,39 @@ def test_avg_signal():
     avg.subscribe(cb)
     sig.put(0)
     assert cb.called
+
+
+def test_unit_conversion_signal():
+    orig = FakeEpicsSignal('sig', name='orig')
+    orig.sim_put(5)
+
+    converted = UnitConversionDerivedSignal(
+        derived_from=orig,
+        original_units='m',
+        derived_units='mm',
+        name='converted',
+    )
+
+    assert converted.original_units == 'm'
+    assert converted.derived_units == 'mm'
+    assert converted.describe()[converted.name]['units'] == 'mm'
+
+    assert converted.get() == 5000
+    converted.put(10000, wait=True)
+    assert orig.get() == 10
+
+    event = threading.Event()
+    cb = Mock()
+
+    def callback(**kwargs):
+        cb(**kwargs)
+        event.set()
+
+    converted.subscribe(callback, run=False)
+    orig.put(20, wait=True)
+    event.wait(1)
+    cb.assert_called_once()
+
+    args, kwargs = cb.call_args
+    assert kwargs['value'] == 20_000
+    assert converted.get() == 20000
```

Comment:
I totally forgot that `20_000` was a thing, but it kind of bugs me you used it on one check and not the other

Reply:
OCD issues resolved in 4110b0e  ;)